### ISSUE TITLE
Build testing issue when a dependecy specifying 'docutils<0.16'

### ISSUE THREAD

**Issue by bitlogik** [Scille/parsec-cloud#1097]
In the setup.py, docutils version is specified to DOCUTILS_DEP = "docutils==0.14".
This might be updated (for the S3 client upgrade to 0.15).

Anyway, when a dependency is specifying docutils<0.16, the build test Azure pipeline for Linux is failing.
For example, botocore added this specification 6 months ago, [in this commit](https://github.com/boto/botocore/commit/add5c8b6b4e208fb0eaba49302df9c3d75b604b9). So when updating it, to fix some issues in older versions (current version is more than 3 y.o.), the Linux test breaks.
This can cause havoc with others dependencies updates in the future.

One cause is that the pip initial installation (in bootstrap step) is installing docutils 0.16
https://github.com/Scille/parsec-cloud/blob/30336b0c94f0993677739a169fe9b91af8bcbc3e/.azure-pipelines.yml#L84 
then on some system, pip/setuptool is cleaning the unnecessary dependencies (enforcing all dependencies constraints) right before _pip check_ (in install step), so the check is OK. But on the Linux Test pipeline, this is not the case, and _pip check_ throws ` botocore 1.15.34 has requirement docutils<0.16,>=0.10, but you have docutils 0.16.`

The Windows pipeline and a manual test on an updated Ubuntu 18.04 performs the cleaning, and removes docutils 0.16, so this is fine. But in the Linux Test pipeline, this is not the case.

I'm not expert in pip and Azure pipeline, but there is something to improve here.
Maybe an old pip version in the Linux pipeline vm ?
Or alternatively force the Linux Boostrap pipeline step to only install DOCUTILS_DEP, enforcing dependencies defined in setup.py.

**Comment by bitlogik** [comment]
The issue can be seen here :
https://dev.azure.com/Scille/parsec/_build/results?buildId=287&view=logs&j=f2f63060-d9d6-52d0-adee-b97db5a9ab91&t=9eed7093-b7bc-5c3d-6995-d4eda620e787

**Comment by bitlogik** [comment]
I tried the "pip update" path to fix this. But nothing works, pip is already updated to a recent version. Probably by UsePythonVersion which loads the latest Python bin tools, or with the "-U" in "pip -U install pip".

There's a tiny detail which prevents the automatic installation of the docutils version 0.16, in some Ubuntu 18.

I reproduced the issue on a Ubuntu 18.04 test machine :
```
apt-get update # update packages addresses
apt-get install -y python3-pip

git clone https://github.com/bitlogik/parsec-cloud.git
cd parsec-cloud
git checkout 385710ca235cb96d88b4342294b0d1b0bea5b0c5

pip3 install -U wheel pip setuptools twine
python3 setup.py bdist_wheel
pip3 install $(ls dist/parsec_cloud-*.whl)[all]
pip3 check parsec[all]
> botocore 1.15.34 has requirement docutils<0.16,>=0.10, but you have docutils 0.16.
```

Finally, this is fixed by adding `pip uninstall -y docutils` before the whl installation.
I see that as a workaround, to a much deeper issue. For example this only applies to docutils, because of botocore requiring docutils<0.16. I feel the same can happen in the future for others packages dependencies when they will need to be updated.

**Comment by touilleMan** [comment]
Using pipdeptree I tracked the conflicting dependencies:
```
boto3==1.12.34
  - botocore [required: >=1.15.34,<1.16.0, installed: 1.15.34]
    - docutils [required: >=0.10,<0.16, installed: 0.16]
[...]
sphinx-intl==2.0.0
  - sphinx [required: Any, installed: 2.4.3]
    - docutils [required: >=0.12, installed: 0.16]
[...]
```
So I guess the issue occurs when pip tries to install sphinx *before* botocore

This issue occurs because unlike poetry, pip doesn't do a full dependency resolution before starting it install :'(

Anyway, I've modified your PR to fix this tell me what you think: f22e58b6ed96977101f77cff80b120ac6606dd5e

**Comment by touilleMan** [comment]
fixed by #1088
